contexts/StatusContext.tsx

```diff
@@ -23,6 +24,25 @@ export const StatusProvider: React.FC<{ children: React.ReactNode }> = ({
   const [statusMessage, setStatusMessage] = useState(user?.activity || "")
   const [friends, setFriends] = useState<string[]>([])
   const [timeSlot, setTimeSlot] = useState("NOW")
+  const [loading, setLoading] = useState(true)
+  const fetchInitialStatus = async () => {
+    try {
+      const response = await api.get("/my-signal")
+      const { status_message, friends, when } = response.data
+
+      setStatusMessage(status_message)
+      setFriends(friends)
+      setTimeSlot(when)
+    } catch (error) {
+      console.error("Error fetching initial activity status:", error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
```

Comment:
is this implementation different from the **fetchMySignal** function that is already existing in **useSignal**?